Source/DafnyLanguageServer/Handlers/DafnyHoverHandler.cs

```diff
@@ -235,12 +236,44 @@ class AssertionBatchIndexComparer : IComparer<AssertionBatchIndex> {
               information += OverCostlyMessage;
             }
           }
+
         }
       }
 
       return information;
     }
 
+    private string GetProofDependencyInformation(IdeState state, TopLevelDeclMemberVerificationTree node, List<AssertionBatchVerificationTree> orderedAssertionBatches) {
+      var allPotentialDependencies = new HashSet<ProofDependency>();
+      var usedDependencies = new HashSet<ProofDependency>();
+      var program = (Program)state.Program;
+      var result = "";
+      if (program is null) {
+        return "null program";
+      }
+
+      var depManager = program.ProofDependencyManager;
+      foreach (var assertionBatch in orderedAssertionBatches) {
+        var batchDependencies = assertionBatch.CoveredIds.Select(depManager.GetFullIdDependency);
+        var batchPotentialDependencies = depManager.GetPotentialDependenciesForDefinition(assertionBatch.VerboseName);
+        allPotentialDependencies.UnionWith(batchPotentialDependencies);
+        usedDependencies.UnionWith(batchDependencies);
+      }
+
+      if (usedDependencies.Any() && options.TrackVerificationCoverage) {
+        result += usedDependencies.Aggregate("\n\nProof dependencies:\n",
```

Comment:
Should we report which parts of the proof were used, if we're already reporting what wasn't used? Isn't that giving redundant information to the user?

I think it's bad if we put too much information in the hover screen, and actually I feel we already have too much in there.

Reply:
It's an experimental interface to display the result of tracking proof dependencies. It won't be activated by default. Only users turning this option for their _project_ on will have this extra information on hover.
We need more return on experience about using this option, how customers use it, what it enables, before taking decisions that will make it to the default UI, assuming we activate this option.
For now, even though it's not the panacea, hover is the cheapest and easiest way to visualize this information, so I suggest we'll keep it that way until we revisit the whole experience of proof dependencies.